src/jit/codegenxarch.cpp

```diff
@@ -718,7 +718,7 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
         // Note that a reg optional operand is a treated as a memory op
         // if no register is allocated to it.
         assert(divisor->isUsedFromReg() || divisor->isMemoryOp() || divisor->IsCnsFltOrDbl() ||
-               divisor->IsRegOptional());
+               divisor->IsRegOptional() || divisor->IsLocal());
```

Comment:
Those are all of the kinds that are allowed, what is left  that is not allowed?
Is there anyway to assert that we don't have that?